### ISSUE TITLE
The GPU memory allocate logic bug in tensorflow 1.15

### ISSUE THREAD

**Issue by alexyinxuanyu** [tensorflow/tensorflow#47807]
<em>
Hello, I tried to upgrade my code from tensorflow 1.12 to tensorflow 1.15. Everything is ok in tf1.12, but I found out some gpu memory allocate logic bug in tf1.15. 

Here are some details: 

1.I fail to use allow_growth=False, tensorflow report "Could not create cudnn handle: CUDNN_STATUS_INTERNAL_ERROR", but in server of my lab, everything is ok, so I guess that maybe it relates to the gpu memory my local computer occupy, e.g. compiz, Xorg. 

2.When i set the allow_growth=True, the code works in tf 1.15, but the gpu memory-usae is so weired, I think gpu memory-usage is linear function of batch_size, but not directly proportional in tensorflow 1.15 which is different from tensorflow 1.12. In tf1.15, the gpu memory-usage is 10G when batchsize=1, 11G when batchsize=2, 19G when batchsize=3, and 17G wen batch size =4. 

In conclusion, I guess that maybe some gpu memory allocate bugs exist in tensorflow 1.15. 

</em>

**System information**
- OS Platform and Distribution: Linux Ubuntu 16.04)
- TensorFlow installed from (source or binary):pip
- TensorFlow version (use command below):tensorflow 1.15.
- Python version:3.6.13
- Bazel version (if compiling from source):1.0.0
- GCC/Compiler version (if compiling from source):7.5.0
- CUDA/cuDNN version:cuda:10 cudnn 7.6.0
- GPU model and memory: 32G tesla V100 (server)and 12G RTX 2080 (local)

**Comment by Saduf2019** [comment]
@alexyinxuanyu 
There is no support for tf 1.x, please upgrade to 2.x and let us know.
[[System requirement](https://github.com/tensorflow/tensorflow/issues/47720#issuecomment-797249831) for the same #47517]
